src/api/db/data/20180214132015_project_log_entry_user_name.rb

```diff
@@ -0,0 +1,14 @@
+class ProjectLogEntryUserName < ActiveRecord::Migration[5.1]
+  def up
+    ProjectLogEntry.where('event_type like "%comment_for%"').each do |log_entry|
+      log_entry.user_name = User.find_by(id: log_entry.user_name).try(:login)
+      log_entry.user_name = '_nobody_' if log_entry.user_name.blank?
```

Comment:
btw: You could do it in a single query if it is too slow

```
    sql = <<-SQL
      UPDATE project_log_entries
      INNER JOIN users ON users.id = project_log_entries.user_name
      SET project_log_entries.user_name = users.login
      WHERE event_type like "%comment_for%"
    SQL
    execute(sql)
```

Reply:
Or at least use in_batches
http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-in_batches

Reply:
You realize this is a data migration right? You can and **should** do it outside of the schema migration.

Reply:
Yes, I do but it will nevertheless lock the table & row while you "migrate".

Reply:
Hm you're right. I'll change it...